### DISCUSSION TITLE
Respect SvelteKit’s `trailingSlash = 'always'` behavior

### DISCUSSION THREAD

**Discussion by danieljb** [jycouet/kitql#693] [Q&A]
Hi, I’ve set up kitql in a project configured to use [trailing slashes](https://kit.svelte.dev/docs/page-options#trailingslash) via export in a `+layout.ts` file.

This generally works, but the links generated by kitql are missing the trailing slash. This can lead to an additional request when SvelteKit redirects to the with-trailing-slash url path, e.g. a link to `/about` will be redirected to `/about/`.

I tried adding the trailing slash to kitql’s route call, e.g. `route('/about/')` but this results in a type error. Is there a way to configure kitql to accept the trailing slash behavior?

**Comment by jycouet** [comment]
Hummm, interesting. Didn't know about this config.

We should probably read this config in the +layout, and apply the correct ending.
I guess that it's applied to all children ? And that it can be switched on/off at any level ?

It's a bit of work, but doable 👍
You want to give a shot ? Or I should start looking at it ?
Let me know

**Reply by danieljb** [reply]
Thanks for the rapid reply! 

Yes, correct. Although the documentation doesn’t explicitly say so, trailing slashes cascade from layout files and I think it’s possible to change configuration on a per-page basis.
Which requires reading the `trailingSlash` values from all of these files: `+(layout|page).ts` as well as `+(layout|page).server.ts`

I’m afraid I’m missing the basic understanding of vite-plugin-kit-routes’ inner workings. Not sure I’ll be able to come up with this functionality.

**Reply by jycouet** [reply]
I just looked a bit more into [trailingslash](https://kit.svelte.dev/docs/page-options#trailingslash).
When I set 
```ts
export const trailingSlash = 'always'
```
I get redirected to `http://localhost:5173/trailingSlash/` (the correct route for me test) but `$page.route.id` is still `"/trailingSlash"`. Could be a bug in SvelteKit ? 

With nesting and all files to check it's kinda lot to do ^^. What's your setup ? You have it in the main layout and want the ending `/` everywhere ? Or it's really case by case ? _(To see how to help you fast)_

**Reply by danieljb** [reply]
In my case I have it set in the main layout and don’t change it anywhere else. I cannot think of a case where an opt-out on a per page base would makes sense.

Come to think of it I could probably just wrap vite-plugin-kit-routes’ `route` in a helper function to append a slash to all but the `/` route? Might be the fastest way to get this running. This probably overlooks some edge cases, but I guess it could work in my current setup.

As for SvelteKit’s `$page.route.id` I could see how this refers to the route definition more than the url path, e.g. if I use (layout groups)[https://kit.svelte.dev/docs/advanced-routing#advanced-layouts-group] these are included in the `id`, too.

**Reply by jycouet** [reply]
Adding a '/' at the end is a bit to simplistic. You can add searchParams in KitRoute and that will not work with your wrapper (unless you go with regex & co)

As you have it globally, I was thinking to add a flag in kitRoute config to add it also at the right place. What do you think ?
(It could be a good first step until someone wants more granularity ?)

**Reply by danieljb** [reply]
Yes, true about the search params.

A global config flag would definitely work in my case. Fully agree this would be a good first step.
